templates/s3-bucket-html-hosting.yaml

```diff
@@ -0,0 +1,90 @@
+AWSTemplateFormatVersion: 2010-09-09
+
+Description: >-
+  This S3 bucket will be used for development and production,
+  and for storing GX reports to be viewable from the Sage AWS VPN.
+
+Parameters:
+
+  BucketName:
+    Type: String
+    Description: Name of the bucket.
+    Default: ''
+
+  EnableVpnAccess:
+    Type: String
+    Description: Whether to grant the Sage VPN read permissions on the bucket.
+    AllowedValues:
+      - "true"
+      - "false"
+    Default: "true"
+
+Conditions:
+  HasBucketName: !Not [!Equals [!Ref BucketName, ""]]
+  EnableVpnAccess:
+    !Equals [!Ref EnableVpnAccess, "true"]
+
+Resources:
+  Bucket:
+    Type: AWS::S3::Bucket
+    DeletionPolicy: Delete
+    Properties:
+      BucketName: !If [HasBucketName, !Ref BucketName, !Ref 'AWS::NoValue']
+      AccessControl: Private
+      OwnershipControls:
+        Rules:
+          - ObjectOwnership: BucketOwnerEnforced
+      BucketEncryption:
+        ServerSideEncryptionConfiguration:
+        - ServerSideEncryptionByDefault:
+            SSEAlgorithm: AES256
+      PublicAccessBlockConfiguration:
+        BlockPublicAcls : true
+        BlockPublicPolicy : true
+        IgnorePublicAcls : true
+        RestrictPublicBuckets : true
+
+  BucketPolicy:
+    Type: AWS::S3::BucketPolicy
+    Properties:
+      Bucket: !Ref Bucket
+      PolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Sid: AccountRead
+            Effect: Allow
+            Principal:
+              AWS:
+                - !Sub '${AWS::AccountId}'
+            Action:
+              - 's3:Get*'
+              - 's3:List*'
+            Resource:
+              - !Sub 'arn:aws:s3:::${Bucket}'
+              - !Sub 'arn:aws:s3:::${Bucket}/*'
+          - !If
+            - EnableVpnAccess
+            - Sid: Allow based on source IP
+              Effect: Allow
+              Principal: '*'
+              Action:
+                - 's3:GetObject'
+              Resource:
+                - !Sub 'arn:aws:s3:::${Bucket}'
+                - !Sub 'arn:aws:s3:::${Bucket}/*'
+              Condition:
+                IpAddress:
+                  aws:SourceIp: '52.44.61.21'
```

Comment:
Where does this value come from?

Reply:
This is the Public IP Address for our AWS VPN defined here: https://sagebionetworks.jira.com/wiki/spaces/IT/pages/1705246745/AWS+Client+VPN+User+Guide